satellite/console/consoleweb/consoleql/reward.go

```diff
@@ -51,8 +57,78 @@ func graphqlReward() *graphql.Object {
 			FieldInviteeCreditDurationDays: &graphql.Field{
 				Type: graphql.Int,
 			},
+			FieldType: &graphql.Field{
+				Type: graphql.Int,
+			},
+			FieldStatus: &graphql.Field{
+				Type: graphql.Int,
+			},
 			FieldExpiresAt: &graphql.Field{
-				Type: graphql.DateTime,
+				Type: graphql.String,
+			},
+		},
+	})
+}
+
+func fromMapRewardInfo(args map[string]interface{}) (reward rewards.OfferInfo, err error) {
+	reward.ID = args[FieldID].(int)
+	if args[FieldInviteeCreditInCents] != nil {
+		reward.InviteeCredit = currency.Cents(args[FieldInviteeCreditInCents].(int))
+	}
+	if args[FieldInviteeCreditDurationDays] != nil {
+		reward.InviteeCreditDurationDays = args[FieldInviteeCreditDurationDays].(int)
+	}
+	if args[FieldAwardCreditInCent] != nil {
+		reward.AwardCredit = currency.Cents(args[FieldAwardCreditInCent].(int))
+	}
+	if args[FieldAwardCreditDurationDays] != nil {
+		reward.AwardCreditDurationDays = args[FieldAwardCreditDurationDays].(int)
+	}
+	reward.RedeemableCap = args[FieldRedeemableCap].(int)
+	if args[FieldType] != nil {
+		reward.Type = args[FieldType].(rewards.OfferType)
+	}
+	if args[FieldStatus] != nil {
+		reward.Status = args[FieldStatus].(rewards.OfferStatus)
+	}
+	expiresAt, err := time.Parse("2006-01-02 15:04:05 -0700 UTC", args[FieldExpiresAt].(string))
```

Comment:
I might be too cautious here but it makes me nervous that we aren't checking for that the type assertions are parseable. The code will panic if we hit that case and if it can happen, Murphy's law is it will.